apps/settings/test/unit/modules/bluetooth/bluetooth_context_test.js

```diff
@@ -0,0 +1,1013 @@
+          name: 'device-01',
+          paired: false
+        };
+        mockPairedDvices = observableArray([
+          {address: '00:11:22:AA:BB:03',
+           name: 'device-03',
+           paired: true}]);
+
+        mockRemoteDevices = observableArray([
+          {address: 'AA:BB:CC:00:11:01',
+           name: 'device-01',
+           paired: false},
+          {address: 'AA:BB:CC:00:11:02',
+           name: 'device-02',
+           paired: false}]);
+
+        mockLenghtOfRemoteDevices = mockRemoteDevices.length;
+        this.sinon.stub(btContext, '_getBluetoothDevice');
+        this.sinon.stub(btContext,
+          'getPairedDevices').returns(mockPairedDvices);
+        this.sinon.stub(btContext,
+          'getRemoteDevices').returns(mockRemoteDevices);
+      });
+
+      test('getPairedDevices() should not be called ' +
+           'getRemoteDevices() should be called ' +
+           '_getBluetoothDevice() should not be called ' +
+           'and no device item created ', function() {
+        btContext._saveDevice(mockFoundDevice);
+        assert.isFalse(btContext.getPairedDevices.called);
+        assert.isTrue(btContext.getRemoteDevices.called);
+        assert.isFalse(btContext._getBluetoothDevice.called);
+        assert.equal(btContext.getRemoteDevices().length,
+          mockLenghtOfRemoteDevices);
+      });
+    });
+  });
+
+  suite('_getBluetoothDevice > ', function() {
+    test('should get same bluetoothDevice from the mock ', function() {
+      btContext._getBluetoothDevice().then(function(bluetoothDevice) {
+        assert.equal(bluetoothDevice, this.mockBluetoothDevice);
+      });
+    });
+  });
+
+  suite('_findDeviceByAddress > ', function() {
+    var address, element;
+    suite('address matched ', function() {
+      setup(function() {
+        address = 'AA:BB:CC:00:11:22';
+        element = {address: 'AA:BB:CC:00:11:22'};
+      });
+
+      test('should return ture ', function() {
+        assert.isTrue(btContext._findDeviceByAddress(address, element));
+      });
+    });
+
+    suite('address does not match ', function() {
+      setup(function() {
+        address = 'AA:BB:CC:00:11:22';
+        element = {address: '00:11:22:AA:BB:CC'};
+      });
+
+      test('should return false ', function() {
+        assert.isFalse(btContext._findDeviceByAddress(address, element));
+      });
+    });
+  });
+
+  suite('pair > ', function() {
+    var mockAdapter, address, mockPromise;
+    setup(function() {
+      mockPromise = new Promise(function(resolve) {
+        resolve();
+      });
+      mockAdapter = {
+        pair: function() {}
+      };
+      address = 'AA:BB:CC:00:11:22';
+      btContext._defaultAdapter = mockAdapter;
+      this.sinon.stub(btContext._defaultAdapter, 'pair').returns(mockPromise);
+    });
+
+    test('_defaultAdapter.pair() should called with address ', function() {
+      btContext.pair(address);
+      assert.isTrue(btContext._defaultAdapter.pair.called);
```

Comment:
`pair` is async, so the assertion should not be made synchronously. If `pair` returns a promise, we can do:
```
btContext.pair(address).then(() => {
  assert.isTrue(btContext._defaultAdapter.pair.called);
  assert.isTrue(btContext._defaultAdapter.pair.calledWith(address));
}).then(done, done);
```